src/H5FDsubfiling/H5FDioc.c

```diff
@@ -1497,8 +1497,10 @@ H5FD__ioc_del(const char *name, hid_t fapl)
 
         /* TODO: No support for subfile directory prefix currently */
         /* TODO: Possibly try loading config file prefix from file before deleting */
-        snprintf(tmp_filename, PATH_MAX, "%s/" H5FD_SUBFILING_CONFIG_FILENAME_TEMPLATE,
-                 prefix_env ? prefix_env : file_dirname, base_filename, (uint64_t)st.st_ino);
+        if (prefix_env && (strlen(prefix_env) > 0)) {
+            snprintf(tmp_filename, PATH_MAX, "%s/" H5FD_SUBFILING_CONFIG_FILENAME_TEMPLATE,
+                     prefix_env ? prefix_env : file_dirname, base_filename, (uint64_t)st.st_ino);
```

Comment:
This `snprintf` call should be ok as it was. This line checks `prefix_env` and adds that in for the first `%s` specifier if the environment variable was set. Otherwise, `file_dirname` is added in. With these changes, not having the environment variable set means nothing will get set in `tmp_filename`, so the open below should always end up failing.

Reply:
My mistake, I was only partially correct here. The `prefix_env ? prefix_env : file_dirname` part inside the snprintf call actually needs to be changed to `prefix_env && (strlen(prefix_env) > 0) ? prefix_env : file_dirname` like you've been doing elsewhere. That way an empty prefix doesn't cause the pathname to be modified to "/original_path" and likely cause failures.